modules/gdscript/gdscript_function.h

```diff
@@ -84,7 +84,7 @@ class GDScriptDataType {
 						Ref<Script> array_script_type_ref = array.get_typed_script();
 
 						if (array_script_type_ref.is_valid()) {
-							valid = (container_element_type->kind == SCRIPT || container_element_type->kind == GDSCRIPT) && container_element_type->script_type == array_script_type_ref.ptr();
+							valid = (container_element_type->kind == SCRIPT || container_element_type->kind == GDSCRIPT) && container_element_type->script_type->cast_to<Script>(array_script_type_ref.ptr());
```

Comment:
`cast_to` is a static function on `Object`, so we always use it as `Object::cast_to<T>(...)`, not calling it on instances.

Reply:
I see. I was trying to check if an array element type is compatible with another element type.